protos/flyteidl/event/event.proto

```diff
@@ -151,13 +151,54 @@ message TaskExecutionEvent {
     // The version field should be incremented when metadata changes across the duration of an individual phase.
     uint32 phase_version                  = 12;    
 
+    // An optional explanation for the phase transition.
+    string reason                         = 13;
+
+
+    // A predefined yet extensible Task type identifier. If the task definition is already registered in flyte admin
+    // this type will be identical, but not all task executions necessarily use pre-registered definitions and this
+    // type is useful to render the task in the UI, filter task executions, etc.
+    string task_type                      = 14;
+
```

Comment:
@katrogan another question, why is task_type / reason not metadata? Maybe a clearer definition in the distinction

Reply:
So I like the documentation at the beginning of this proto message: `Plugin specific execution event information`  i.e. attributes of the event itself. In this case task_type is an attribute of the execution and reason goes along with phase and phase version as part of the actual event update, right? I see metadata is more for identifying/discerning how a task was executed rather than artifacts or products of the event itself.